src/components/DailyStatistics/utils.ts

```diff
@@ -0,0 +1,39 @@
+import { groupBy, forEach } from "lodash";
+import { DailyStatistics, CampaignPolicy } from "../../types";
+
+export const summariseTransactions = (
+  response: DailyStatistics,
+  policies: CampaignPolicy[] | null
+): {
+  summarisedTransactionHistory: {
+    name: string;
+    category: string;
+    quantity: number;
+  }[];
+  summarisedTotalCount: number;
+} => {
+  const summarisedTransactionHistory: {
+    name: string;
+    category: string;
+    quantity: number;
+  }[] = [];
+  let summarisedTotalCount = 0;
+
+  const transactionsByCategory = groupBy(response.pastTransactions, "category");
+  forEach(transactionsByCategory, (value, key) => {
+    let quantityPerCategory = 0;
+    transactionsByCategory[key].forEach(transaction => {
+      quantityPerCategory += transaction.quantity;
+    });
+    summarisedTotalCount += quantityPerCategory;
+    const itemWithName = policies?.find(item => item.category === key);
+
+    summarisedTransactionHistory.push({
+      name: itemWithName?.name ?? key,
+      category: key,
+      quantity: quantityPerCategory
+    });
+  });
```

Comment:
We can use a `reduce` here instead, rather than `lodash`'s forEach. Also, we can make use of `lodash` methods to  make this cleaner:
```
import { chain, sumBy } from "lodash";
import { DailyStatistics, CampaignPolicy } from "../../types";

type SummarisedTransactions = {
  summarisedTransactionHistory: {
    name: string;
    category: string;
    quantity: number;
  }[];
  summarisedTotalCount: number;
};

export const summariseTransactions = (
  response: DailyStatistics,
  policies: CampaignPolicy[] | null
): SummarisedTransactions => {
  return chain(response.pastTransactions)
    .groupBy("category")
    .reduce<SummarisedTransactions>(
      (prev, transactionsByCategory, key) => {
        const totalQuantityInCategory = sumBy(
          transactionsByCategory,
          "quantity"
        );
        prev.summarisedTransactionHistory.push({
          name: policies?.find(item => item.category === key)?.name ?? key,
          category: key,
          quantity: totalQuantityInCategory
        });
        prev.summarisedTotalCount += totalQuantityInCategory;
        return prev;
      },
      { summarisedTransactionHistory: [], summarisedTotalCount: 0 }
    )
    .value();
};
```

Reply:
thanks seb! for this case as well, tho unlikely, if the item that is transacted does not belong in the policies, then leaving the `name=category` is fine?

Reply:
Yea good point, I think it's a sensible default fallback.